### PR TITLE
Make config files built in at build

### PR DESCRIPTION
**Pull request by rminnich**
We will later allow overrides, but this means that you
can go get wingo and run it without having to install a bunch
of files all over the place as root.

Create map of file paths to []byte
If we can't open
a file we can check in the FileMap to see if it's already there.
We use the FileMap for default scripts, data, PNGs, and so on.
If a file is not found, we can fall back to the built in file.

As long as we're at it we save them in there if we open them.
This saves all the many times it opens TTF files.

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

### CODE REVIEW

File: wm/config.go [hunk 1]
```diff
@@ -120,13 +120,15 @@ func (conf *Configuration) loadMouseConfigSection(
...
 		for _, cmd := range key.Strings() {
+			logger.Warning.Printf("mouse cmd is %v", cmd)
 			if _, ok := conf.mouse[ident]; !ok {
 				conf.mouse[ident] = make([]mouseCommand, 0)
 			}
 
 			if err := gribbleEnv.Check(cmd); err != nil {
 				logger.Warning.Printf(
 					"Could not parse command '%s' because: %s", cmd, err)
+				panic("fuck")
```
Comment (GanShun) [inline wm/config.go]: Something more descriptive please =P

Comment (GanShun) [reply wm/config.go]: do you want to change the warning printf into some kind of fatal instead?

File: wm/config.go [hunk 2]
```diff
@@ -147,13 +149,15 @@ func (conf *Configuration) loadKeyConfigSection(
...
 		for _, cmd := range key.Strings() {
+			logger.Warning.Printf("cmd is %v", cmd)
 			if _, ok := conf.key[section]; !ok {
 				conf.key[section] = make([]keyCommand, 0)
 			}
 
 			if err := gribbleEnv.Check(cmd); err != nil {
 				logger.Warning.Printf(
 					"Could not parse command '%s' because: %s", cmd, err)
+					panic("key")
```
Comment (GanShun) [inline wm/config.go]: likewise as above

File: wm/config.go (hunk 1 above)
Comment (ryukinix) [reply wm/config.go]: Seems a good case to use a kind of `log.Fatalf` here.

**Comment by rminnich** [comment]
BurntSushi used panic EVERYWHERE. I don't want to fix that just now. I just removed those calls to panic ... but panic is the wingo style at present.

File: commands/commands.go [hunk 3]
```diff
@@ -226,6 +226,7 @@ Make sure "audio_play_cmd" is set to a program that can play wav files.
 }
 
 func (cmd Dale) Run() gribble.Value {
+	return nil
```
Comment (bkochendorfer) [inline commands/commands.go]: Perhaps add a comment about why you nerfed this function? I see the related `# audio_play_cmd` and possibly `AudioProgram:    "", //"aplay",` so just curious why you also had to return early here.

File: wm/config_theme_setters.go [hunk 4]
```diff
@@ -160,23 +166,31 @@ func setGradient(k wini.Key, clr *render.Color) {
...
 }
 
 func setFont(k wini.Key, place **truetype.Font) {
 	if v, ok := getLastString(k); ok {
-		bs, err := ioutil.ReadFile(v)
+		bs, err := misc.DataFile(v)
 		if err != nil {
 			logger.Warning.Printf(
 				"Could not get font data from '%s' because: %v", v, err)
+			panic("font")
```
Comment (bkochendorfer) [inline wm/config_theme_setters.go]: I saw the related reviews about `panic` and wanted to point one more out here but if you left it in intentionally don't worry about it.

File: main.go [hunk 5]
```diff
@@ -161,7 +161,8 @@ func main() {
 	manageExistingClients()
 
 	// Now make sure that clients are in the appropriate visible state.
-	for _, wrk := range wm.Heads.Workspaces.Wrks {
+	for i, wrk := range wm.Heads.Workspaces.Wrks {
+		logger.Warning.Printf("Window %d is visible? %v", i, wrk.IsVisible())
```
Comment (bkochendorfer) [inline main.go]: Might be better suited as `logger.Message`

Comment (bkochendorfer) [inline main.go]: Might be better suited as `logger.Message`.

File: commands/commands.go (hunk 3 above)
Comment (rminnich) [reply commands/commands.go]: it has to die. He says it's an easter egg and I see no need of such things. This whole pile of code needs a lot of love

File: main.go (hunk 5 above)
Comment (rminnich) [reply main.go]: oh dear, this needs to be gone, hang on ...

File: wm/config_theme_setters.go (hunk 4 above)
Comment (rminnich) [reply wm/config_theme_setters.go]: Fixed, thanks.

**Comment by rminnich** [comment]
Note I added the output of go generate (the datafiles.go file) to make go get work. They really just about never change, and I've notice in a lot of the upstream go source, even when a go generate is done, they included the output of the go generate in the repo.

**Review by bkochendorfer — APPROVED** [review]

**Pull request merged**